src/opcodes/flow.cairo

```diff
@@ -76,3 +76,33 @@ pub fn opcode_verify(ref engine: Engine) -> Result<(), felt252> {
 pub fn opcode_return(ref engine: Engine) -> Result<(), felt252> {
     return Result::Err('opcode_return: returned early');
 }
+
+pub fn opcode_push(opcode: u8, script: @ByteArray, mut opcode_idx: usize) -> usize {
+    let mut result: usize = 0;
+    if opcode == Opcode::OP_PUSHDATA1 {
+        let nextop_code = script[opcode_idx + 1];
+        let opcode_u32: u32 = nextop_code.into();
+        opcode_idx += opcode_u32 + 2; // return this
+        result = opcode_idx;
+    } else if opcode == Opcode::OP_PUSHDATA2 {
+        let nextop_code1 = script[opcode_idx + 1];
+        let nextop_code2 = script[opcode_idx + 2];
+        let opcode_u32_1: u32 = nextop_code1.into();
+        let opcode_u32_2: u32 = nextop_code2.into();
+        opcode_idx += opcode_u32_1 + opcode_u32_2 + 3;
+        result = opcode_idx;
+    } else if opcode == Opcode::OP_PUSHDATA4 {
+        let nextop_code1 = script[opcode_idx + 1];
+        let nextop_code2 = script[opcode_idx + 2];
+        let nextop_code3 = script[opcode_idx + 3];
+        let nextop_code4 = script[opcode_idx + 4];
+        let opcode_u32_1: u32 = nextop_code1.into();
+        let opcode_u32_2: u32 = nextop_code2.into();
+        let opcode_u32_3: u32 = nextop_code3.into();
+        let opcode_u32_4: u32 = nextop_code4.into();
+        opcode_idx += opcode_u32_1 + opcode_u32_2 + opcode_u32_3 + opcode_u32_4 + 5;
```

Comment:
This isn't really accurate to how these opcode lengths work. You should be able to use `pull_data` from the engine to get the correct amount to increase the opcode index by.
https://github.com/keep-starknet-strange/shinigami/blob/9f5af65c750c6c420959206ac6850ebb36210cef/src/engine.cairo#L83

Here is an example of how it is used to get the data length in pulldata_x opcodes :
https://github.com/keep-starknet-strange/shinigami/blob/9f5af65c750c6c420959206ac6850ebb36210cef/src/opcodes/constants.cairo#L17

Reply:
Hi @b-j-roberts  I just uploaded a new commit, in this commit I rename the file src/opcodes/utils to src/opcodes/utils_opcodes.cairo to avoid confusion with the src/utils.cairo.
Also I use the pull_data function and refactor the implementation in the Engine 

Please review and let me know if I have to change something.